web/src/test/java/com/graphhopper/http/resources/CustomWeightingRouteResourceTest.java

```diff
@@ -85,18 +85,22 @@ public void testAvoidArea() {
         JsonNode yamlNode = clientTarget(app, "/custom").request().post(Entity.entity(yamlQuery,
                 new MediaType("application", "yaml"))).readEntity(JsonNode.class);
         JsonNode path = yamlNode.get("paths").get(0);
+        // todonow: why use assertBetweeen?
         assertBetween("distance wasn't correct", path.get("distance").asDouble(), 500, 900);
+        //assertEquals(path.get("distance").asDouble(), 661, 10);
 
         yamlQuery += "priority:\n" +
-                "  area_custom1: 0.5\n" +
+                // todonow: the polygon used here is a bit weird and we really have to use a low priority to force the
+                // large detour
+                "  area_custom1: 0.05\n" +
```

Comment:
Here it looks like you were trying to enforce the green route instead of the red one using the area shown as grey box:

![image](https://user-images.githubusercontent.com/17603532/80457282-54f48280-892f-11ea-9c4c-e56e3b0d804b.png)

But this seems a bit strange to me because both routes are crossing the area. With the new formula I was not able to enforce this route and instead it yields a route that really avoids the grey box (goes around it and therefore the distance is much longer)

Reply:
Maybe this is exactly because the priority contribution depends on the speed: The green route is probably faster and therefore decreasing the priority does not increase the weighting as much as on the red route.

Reply:
Hmmh, ugly. I'll have a look why I chose this area and then accepted that the path was still going thought (might be the intention but I'm no longer 100% sure)

Reply:
This seems like a key example to me: We set the priority to some low value in this area and with the old formula this means we add a bigger penalty to the slower road and a smaller penalty to the faster road. Therefore we have this situation where one of the roads crossing the area is surpressed, but the other is not. With the new formula we apply the *same* penalty to both roads (per meter) and therefore they are both neglected. What is better? I don't know :) I guess that depends on what you want to achieve.

Reply:
The old formula also needs a 0.05 priority to force the long detour, but yes, for all values until 0.05 the old formula prefers the longer way where the new prefers the shorter. So here the new formula makes more sense as inside the avoided area it influences the original path the least.

And yes, we should remove assertBetween and replace with assertEquals and a bigger delta.